### PR TITLE
[pt1][quant] Move _th_fill_ to ATen and implement that for QInt8

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #18546 [pt1][quant] Add Quantized Backend&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D14637671/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #18814 [pt1][quant] Add slicing and int_repr() to QTensor&nbsp;&nbsp;[:yellow_heart:](https://our.internmc.facebook.com/intern/diff/D14756833/)
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#19211 [pt1][quant] Move _th_fill_ to ATen and implement that for QInt8**&nbsp;&nbsp;[:yellow_heart:](https://our.internmc.facebook.com/intern/diff/D14781825/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #18960 [pt1][quant] Add empty_quantized&nbsp;&nbsp;[:yellow_heart:](https://our.internmc.facebook.com/intern/diff/D14810261/)

Implement _th_fill for QInt8 so we can assign to QTensor
```
r = torch.ones(1, dtype=torch.float)
scale = 1
zero_point = 2
qr = r.quantize_linear(scale, zero_point)
qr[0] = 8 # 8 will be quantized and then assigned to q.
# qr.item() == 8
```

Differential Revision: [D14781825](https://our.internmc.facebook.com/intern/diff/D14781825/)

### CODE REVIEW

File: aten/src/ATen/Dispatch.h [hunk 1]
```diff
@@ -159,6 +159,7 @@ inline void deprecated_AT_DISPATCH_ALL_TYPES_AND_HALF_AND_COMPLEX() {}
       AT_PRIVATE_CASE_TYPE(at::ScalarType::Int, int32_t, __VA_ARGS__)        \
       AT_PRIVATE_CASE_TYPE(at::ScalarType::Long, int64_t, __VA_ARGS__)       \
       AT_PRIVATE_CASE_TYPE(at::ScalarType::Short, int16_t, __VA_ARGS__)      \
+      AT_PRIVATE_CASE_TYPE(at::ScalarType::QInt8, uint8_t, __VA_ARGS__)      \
```
Comment (gchanan) [inline aten/src/ATen/Dispatch.h]: don't do this -- I know it's called ALL_TYPES, but it means something like ALL_STANDARD_TYPES.  The issue is that extensions use this and you will break them if you add another type.

File: aten/src/ATen/core/Tensor.h [hunk 2]
```diff
@@ -177,6 +177,26 @@ class CAFFE2_API Tensor {
     return impl_->is_contiguous();
   }
 
+  bool is_transposed() const {
```
Comment (gchanan) [inline aten/src/ATen/core/Tensor.h]: is this used?

**Comment by jerryzh168** [comment]
This is not done yet plz don't review yet.
This diff should be dependent on Christian's diff, I'll refresh this after that is done

**Comment by jerryzh168** [comment]
will branch on scalartype for now since @cpuhrsch 's pr is getting more complicated and won't land soon.

**Pull request closed without merging**